at_client/lib/src/util/at_client_validation.dart

```diff
@@ -51,4 +66,44 @@ class AtClientValidation {
       throw AtKeyException('$atSign does not exist');
     }
   }
+
+  /// Validates the atKey.
+  /// Throws [UnAuthorizedException] if atKey is a cached one.
+  static Future<void> validateAtKey(AtKey atKey, {AtClientPreference? preferences}) async {
+    _validateKey(atKey.key!);
+    _validateNamespace(atKey.namespace);
+    if ((atKey.metadata != null && atKey.metadata!.isCached) || atKey.key!.startsWith('cached:')) {
```

Comment:
Please add a condition to avoid caching of self keys

if sharedBy equals currentAtSign and isCaching is true, throw exception.